### ISSUE TITLE
Display of tasks belonging to multiple list broken

### ISSUE THREAD

**Issue by morgonhed** [mpcjanssen/simpletask-android#308]
Consider this todo-list:

    a @a
    b @b
    ab @a @b 

When displayed without filter it looks like this:

    A
       a
       ab
    B
       b

This is not intuitive as the task ab belongs to both lists and therefore (that's my intuition at least) should be displayed in both lists.

It gets worse when one filters by list. This is what it looks like when filtered by list "b":

    A
       ab
    B
       b

So even though the filter ist set to only display list b, list a shows up. To my intuition the following would be the expected behavior in this case:

    B
       b
       ab

**Comment by mpcjanssen** [comment]
To start, this is working as designed so it's not broken in that sense. This is all a consequence of tasks being on more than one list. If you filter for a single list and then sort by lists, you got exactly what you asked for.

Now whether the design is broken or not is up for discussion. 

Firstly it's currently not very easy to show a single task more than once in the UI. I don't think this is fixable without changing the backend code to support actual queries instead of a simple list.

Secondly only showing tasks in list or tags which are in the filter is a much harder problem than it seems. If I have task `a +b +c +d` and the tag filter is `b` or `c`, where should the task be displayed? And how does that change if `b` is deselected? According to what you suggest it first should be under tag `b` and then after `b` is deselected under `c`. 
So this requires parsing of the current filter to decide where to display a task.

So short answer this is probably not going to be changed anytime soon.

**Comment by smichel17** [comment]
Would it be possible to implement a partial solution via a special case? If you're sorting by list, and you filter by a single list, display all visible items under the header of that list.

This would fix the second original example (which I agree is the more obviously unintuitive behavior) without getting into the complexity of showing a single task multiple times.

**Comment by mpcjanssen** [comment]
@smichel17 this would be relatively easy (I think) I will look into this.

**Comment by smichel17** [comment]
#160

**Comment by mpcjanssen** [comment]
Not really related to #122. #122 is about the display of a single task. This is about the display of a whole tasklist. This is very hard to fix.

**Comment by smichel17** [comment]
I think a true fix for this would depend on #153.

**Comment by mpcjanssen** [comment]
Duplicate of #153
